Makefile

```diff
@@ -21,23 +21,27 @@ endif
 download:
 	go mod download
 
-# Note for linux users running this command, should be ran inside of a nix shell
-#  for example: nix develop -c make install
 install:
 ifeq ($(OSFLAG),$(WINDOWS))
-		echo "If you are running windows and know how to install what is needed, please contribute by adding it here!"
-		exit 1
+	echo "If you are running windows and know how to install what is needed, please contribute by adding it here!"
+	exit 1
 endif
 ifeq ($(OSFLAG),$(OSX))
-		brew install asdf
-		asdf plugin-add golang https://github.com/kennyp/asdf-golang.git || true
-		asdf plugin-add nodejs https://github.com/asdf-vm/asdf-nodejs.git || true
-		asdf plugin-add ginkgo https://github.com/jimmidyson/asdf-ginkgo.git || true
-		asdf install
+	brew install asdf
+	asdf plugin-add nodejs https://github.com/asdf-vm/asdf-nodejs.git || true
+	asdf plugin-add rust https://github.com/code-lever/asdf-rust.git || true
+	asdf plugin-add golang https://github.com/kennyp/asdf-golang.git || true
+	asdf plugin-add ginkgo https://github.com/jimmidyson/asdf-ginkgo.git || true
+	asdf plugin-add pulumi || true
+	asdf install
 endif
 ifeq ($(OSFLAG),$(LINUX))
-		# pulls the ginkgo version from the .tool-versions asdf file to install so we only have the version in one place
-		go install github.com/onsi/ginkgo/v2/ginkgo@v$(shell cat ./.tool-versions | grep ginkgo | sed -En "s/ginkgo.(.*)/\1/p")
+	# install nix if not running in ci since we use a github action there
+ifneq ($(IS_CI),true)
+	sh <(curl -L https://nixos-nix-install-tests.cachix.org/serve/vij683ly7sl95nnhb67bdjjfabclr85m/install) --daemon --tarball-url-prefix https://nixos-nix-install-tests.cachix.org/serve --nix-extra-conf-file ./nix.conf
+endif
+	# pulls the ginkgo version from the .tool-versions asdf file to install so we only have the version in one place
```

Comment:
Not sure I understand why do we do it like this when [a `ginkgo` plugin for NIX is available](https://github.com/NixOS/nixpkgs/blob/nixos-21.11/pkgs/development/tools/ginkgo/default.nix)?

If it's just to have a version declared in one place, well other versions are also currently duplicated between `shell.nix` & `.tools-versions` 🤷

Reply:
It just really sucks that we need to put stuff in two places. This one needs to be the same version whereas things like the go version and node version won't care if they are one . version off for us locally vs in ci. I will update it to use the plugin, I will just put a very heavy comment about making sure they are both the same in both places.